code/modules/economy/holopay.dm

```diff
@@ -198,24 +194,9 @@
  * Deletes the holopay thereafter.
  */
 /obj/structure/holopay/proc/dissapate()
-	var/obj/item/card/id/linked_card = get_card()
 	playsound(loc, "sound/effects/empulse.ogg", 40, TRUE)
 	visible_message(span_notice("The pay stand vanishes."))
-	QDEL_NULL(linked_card.holopay_ref)
-
-/**
- * Checks that the card is still linked.
- * Deletes the holopay if not.
- *
- * Returns:
- * * /obj/item/card/id/card - The card that is linked to the holopay
- */
-/obj/structure/holopay/proc/get_card()
-	var/obj/item/card/id/linked_card = card_ref?.resolve()
-	if(!linked_card || !istype(linked_card, /obj/item/card/id))
-		stack_trace("Could not link a holopay to a valid card.")
-		qdel(src)
-	return linked_card
+	QDEL_NULL(linked_card.my_store)
```

Comment:
rather then clearing this in dissipate with QDEL_NULL, you should do it in Destroy. That way you actually set linked_card to cull, and clear the ref